### ISSUE TITLE
Text styles positions should update with character deletions

### ISSUE THREAD

**Issue by emmnull** [michael/svedit#15]
Gave a quick try to the demo site and it appears that text styles (bold, italic, underline, etc.) keep their original character-index position when deleting characters, causing them to shift from the original strings they were applied to. This does not appear to occur when inserting content, was only observed when removing content (backspace, ctrl-x).

## Repro

1. Go to demo site
2. Apply any text style to a sequence of characters
3. Remove characters before the styled content
4. Observe styles shifting away from characters they were originally applied to

## Expected behavior

Text styling positions should be attached to character sequences and behave as they do when inserting new content causes content to shift.

**Comment by michael** [comment]
True, it works for expansion of annotations, but not the deletion. Will look into it asap.

Thanks for reporting!

**Comment by michael** [comment]
Took a while, but fixed now.

https://github.com/michael/svedit/commit/948826e57d55e6213b83fa8cae5d6dfc3031c723

Will dedicated all summer to working on Svedit. Fingers crossed we end up with something ready for production!

Current progress is at: https://github.com/michael/svedit/pull/27
